inc/cobs.h

```diff
@@ -0,0 +1,71 @@
+#pragma once
+
+#include <cstdint>
+#include <cstddef>
+
+/*
+Consistent Overhead Byte Stuffing (COBS) removes a specific value from a
+list of values. The removed value can then, for example, be used as a
+frame marker.
+
+Refer to: https://en.wikipedia.org/wiki/Consistent_Overhead_Byte_Stuffing
+
+COBS considerations:
+
+1. Configurable marker byte value. The current implementation assumes
+0x00.
+
+2. If the encoding results in { ..., 0, 1 }, encode it as { ..., 0 }.
+The current implementation applies this reduced encoding.
+
+3. If the encoding results in { n, ..., x, 0 } where n is an offset and
+x > n, encode it as { x, ..., 0 }. The standard implementation would run
+into an decoding error because x > n is bigger than the remaining number
+of bytes. The current implementation does not apply this reduced
+encoding.
+
+4. Append the marker byte to the encoded result. The current
+implementation does append the marker byte.
```

Comment:
Fair enough. However, this encoding scheme is primarily being used for unambiguous packet framing for serial data. This means the use case will practically always require appending the marker byte. Again, what is the use case for doing a COBS encode without appending the marker byte? There are many other encoding schemes that result in reduced byte size or include redundancy/error correction and those would certainly be better choices if you don't care about streaming data.

For our case, when we use cobs::encode on the microcontroller, we will _always_ append a marker byte to the end of the buffer as the encoded data will then be passed to a low level driver that actually handles the transmission.

Unless there is a specific use case that benefits from excluding the marker byte, we should append it as that is how the algorithm is normally described and implemented.